curated_transformers/util.py

```diff
@@ -43,3 +48,81 @@ def all_equal(iterable):
     (or if the input is an empty sequence), False otherwise."""
     g = itertools.groupby(iterable)
     return next(g, True) and not next(g, False)
+
+
+def gradual_transformer_unfreezing_per_pipe(
+    nlp: Language,
+    callback_args: Dict[str, Any],
+    freeze_params: Dict[str, int],
+    last_unfreeze_step: int,
+):
+    current_step = callback_args["step"]
+    if current_step > last_unfreeze_step:
```

Comment:
`last_unfreeze_step` makes the code less readable. Is this optimization really needed?

If it is, I'd do the `max`-ing here, since it is really an implementation detail of the function.

Reply:
Probably not, to be honest. I think we'll be fine if it's removed.